### ISSUE TITLE
[BUG] PLAID Dataset is being loaded as unequal always

### ISSUE THREAD

**Issue by hadifawaz1999** [aeon-toolkit/aeon#2224]
### Describe the bug

When loading the PLAID dataset (originally unequal length) with `load_classification` it is always loading the list version  never the equal length version

### Steps/Code to reproduce the bug

```python
x, y = load_classification(name="PLAID")
print(x.shape)
```

### Expected results

shape output

### Actual results

Attribute error, list does not have attribute shape

**Comment by lucifer4073** [comment]
Hi @hadifawaz1999, I would like to work on this issue. Can you please assign it to me?

**Comment by MatthewMiddlehurst** [comment]
@lucifer4073 Please have a look through the contributing guide. This issue involved loading external data so theres a chance you wont be able to fix this.

**Comment by lucifer4073** [comment]
Hi @MatthewMiddlehurst, thank you for your response. I attempted to debug the issue and found that x has an inhomogeneous shape, which prevents it from being converted into a NumPy array. This seems to stem from the external data being loaded.

```
Type of x -> <class 'list'>
Type of y -> <class 'numpy.ndarray'>
Type of x[0] -> <class 'numpy.ndarray'>
Type of x[0][0] -><class 'numpy.ndarray'>
```
I’ve attached this for reference in case someone else wants to review or provide further insights. 

Attempt: `x=np.asarray(x)`

Result:  `ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (1074, 1) + inhomogeneous part.`

**Comment by MatthewMiddlehurst** [comment]
The function has a `load_equal_length` parameter which should handle this, but I don't think its an issue with the function itself.

**Comment by hadifawaz1999** [comment]
@TonyBagnall what do you think

**Comment by saadaltohamy** [comment]
Hi, I found that the Plaid dataset doesn't have _disc, _eq, or _nmv versions. So when calling _load_saved_dataset(), it returns a list of 2D arrays due to variable-length multivariate series.

> Shape of the nested list:
> (1074,)
> Shapes of the first 5 elements in X:
> (1, 500)
> (1, 500)
> (1, 500)
> (1, 200)
> (1, 544)

Trying to convert this list to a 3D NumPy array with _from_np_list_to_numpy3d() raises:
ValueError: Cannot convert unequal length to numpy3D.

I think we could handle this by padding sequences when this error is raised. If that sounds good, I’m happy to open a PR to add a padding utility. @hadifawaz1999 @MatthewMiddlehurst @TonyBagnall

**Comment by MatthewMiddlehurst** [comment]
No. Automatically padding data like this is not something we want to do IMO.

**Comment by hadifawaz1999** [comment]
yeah i agree with @MatthewMiddlehurst it is actually an issue from the timeseriesclassification.com website @TonyBagnall probably the equal length version is not uploaded. Plus if am  not mistaken (correct me @MatthewMiddlehurst @TonyBagnall ) the equal length version is not padding zero, it is padding the mean value per series followed by adding small gaussian noise on the padding places, this was Eamonn's solution on the ucr website, so i assumed you did the same.

**Comment by saadaltohamy** [comment]
Is it allowed for the function to return datasets with variable-length time series? e.g., list of ndarrays that have variable length 
@MatthewMiddlehurst @hadifawaz1999

**Comment by saadaltohamy** [comment]
As I see in docstring, I think the answer is yes. 

So yeah, we don't have a bug here, at least in the function.

**Comment by TonyBagnall** [comment]
we reviewed this and know why this happens only for PLAID. The function load_classification is a bit convoluted because it does several things for convenience. If you just want to directly load a dataset, use load_from_ts_file. 

It first looks in extract_path to see if the file is already on disk. This defaults to datasets/data, so we can easily load baked in datasets using load_classification. If it is not there it creates the directory, downloads and unpacks from tsc.com.

For the load equal length versions

```
x, y = load_classification(name="PLAID", load_equal_length=True)
```
it looks in the directory (downloaded or not), to see if the problem with the extension _eq is in the file. This is in the downloaded zip if the standard is not equal length. However, the equal it is NOT in the PLAID shipped version. So it just loads the unequal length one.

So, to solve we considered:
1. Ship the equal length ts file in aeon: dont want to do this because of the size of the file
2. Dont look in datasets directory, always download if not in the default dir. I dont like this because we use load_classification in examples, and it will result in a lot of unnecessary downloads and fails for internet connections
3. Replace PLAID with a much smaller unequal length problem, and ship with the equal length equivalent

I prefer 3, as part of a overhaul of the shipped data.

**Comment by TonyBagnall** [comment]
this is fixed by 2859

```python
x, y = load_classification(name="PLAID", load_equal_length=True)
print(x.shape)
```
(1074, 1, 1345)

although make sure you delete the PLAID directory in datasets.data if you are not working from a clean install, it did not get automatically deleted from mine

<img width="193" height="262" alt="Image" src="https://github.com/user-attachments/assets/12676300-b3f9-4d19-a039-02a3586e79a9" />
